src/charting/show-dashboard.js

```diff
@@ -53,83 +53,162 @@ function __uploadPNG__(robot, room, fileName) {
-    if (splits2 === 0) {
-      dataset2.push( { label: "No Hardware", count: 0, percent: 100} );
-    } else {
-      let slices2 = [];
-      for (let i = 0; i < splits2; i++) {
-        slices2.push(aggregatedServerHardware[0].counts[i]);
-        totalCount2 += aggregatedServerHardware[0].counts[i].count;
-      };
-      for (let j = 0; j < splits2; j++) {
-        dataset2.push({ label: slices2[j].value, count: (slices2[j].count), percent: (slices2[j].count / totalCount2) });
-      };
-    }
+  return { splits: splits1, totalCount: totalCount1, dataset: dataset1 };
+}
 
-    //separate data for alerts
-    let splits3 = aggregatedAlerts[0].counts.length;
-    let dataset3 = [];
-    let totalCount3 = 0;
-    if (splits3 === 0) {
-      dataset3.push({ label: "No Alerts", count: 0, percent: 100} );
-    }
-    else {
-      let slices3 = [];
-      for (let i = 0; i < splits3; i++) {
-        slices3.push(aggregatedAlerts[0].counts[i]);
-        totalCount3 += aggregatedAlerts[0].counts[i].count;
-      };
-      for (let j = 0; j < splits3; j++) {
-        dataset3.push({ label: slices3[j].value, count: (slices3[j].count), percent: (slices3[j].count / totalCount3) });
-      };
-    }
+function __transformHardwareData__(aggregatedServerHardware) {
+  let splits2 = aggregatedServerHardware[0].counts.length;
+  let dataset2 = [];
+  let totalCount2 = 0;
+  if (splits2 === 0) {
+    dataset2.push( { label: "No Hardware", count: 0, percent: 100} );
+  } else {
+    let slices2 = [];
+    for (let i = 0; i < splits2; i++) {
+      slices2.push(aggregatedServerHardware[0].counts[i]);
+      totalCount2 += aggregatedServerHardware[0].counts[i].count;
+    };
+    for (let j = 0; j < splits2; j++) {
+      dataset2.push({ label: slices2[j].value, count: (slices2[j].count), percent: (slices2[j].count / totalCount2) });
+    };
+  }
+
+  return { splits: splits2, totalCount: totalCount2, dataset: dataset2 };
+}
+
+function __transformAlertsData__(aggregatedAlerts) {
+  let splits3 = aggregatedAlerts[0].counts.length;
+  let dataset3 = [];
+  let totalCount3 = 0;
+  if (splits3 === 0) {
+    dataset3.push({ label: "No Alerts", count: 0, percent: 100} );
+  }
+  else {
+    let slices3 = [];
+    for (let i = 0; i < splits3; i++) {
+      slices3.push(aggregatedAlerts[0].counts[i]);
+      totalCount3 += aggregatedAlerts[0].counts[i].count;
+    };
+    for (let j = 0; j < splits3; j++) {
+      dataset3.push({ label: slices3[j].value, count: (slices3[j].count), percent: (slices3[j].count / totalCount3) });
+    };
+  }
+
+  return { splits: splits3, totalCount: totalCount3, dataset: dataset3 };
+}
 
-    //separate data for hardware with profiles
-    let splits4 = aggregatedHardwareWithProfiles[0].counts.length;
-    let dataset4 = [];
-    let totalCount4 = 0;
-    if (splits4 === 0) {
-      dataset4.push({ label: "No Hardware", count: 0, percent: 100} );
+function __transformHardwareWithProfilesData(aggregatedHardwareWithProfiles) {

```

Comment:
in this function i would rename the local variables to drop the '4' from the end.